src/app/services/courses-store.service.ts

```diff
@@ -1,44 +1,162 @@
+      },
+      error: err => {
+        this.isLoading$$.next(false);
+        console.error('Failed to add course', err);
+      },
+    });
   }
 
-  getCourse(id: string) {
-    // Add your code here
+  getCourse(id: string): Observable<CourseResponse> {
+    this.isLoading$$.next(true);
+    return this.courseService.getCourse(id).pipe(
+      tap({
+        next: () => {
+          this.isLoading$$.next(false);
+        },
+        error: err => {
+          console.error('Failed to get course', err);
+          this.isLoading$$.next(false);
+        },
+      })
+    );
   }
 
-  editCourse(id: string, course: any) {
-    // replace 'any' with the required interface
-    // Add your code here
+  editCourse(id: string, course: Omit<Course, 'id' | 'creationDate'>) {
+    this.isLoading$$.next(true);
+    this.courseService.editCourse(id, course).subscribe({
+      next: () => {
+        this.courses$$.next(
+          this.courses$$.value.map(courseElement => {
+            if (courseElement.id === id) return { ...course, id: id };
+            else return courseElement;
+          })
+        );
+        this.isLoading$$.next(false);
+      },
+      error: err => {
+        this.isLoading$$.next(false);
+        console.error('Failed to edit course', err);
+      },
+    });
   }
 
   deleteCourse(id: string) {
-    // Add your code here
+    this.isLoading$$.next(true);
+    this.courseService.deleteCourse(id).subscribe({
+      next: (response: DeleteCourseResponse) => {
+        if (response.successful) this.courses$$.next(this.courses$$.value.filter(course => course.id !== id));
+      },
+      error: err => {
+        this.isLoading$$.next(false);
+        console.error('Failed to delete course', err);
+      },
+    });
   }
 
-  filterCourses(value: string) {
-    // Add your code here
+  filterCourses(filterInfo: FilterCourse): Observable<FilterResponse> {
+    this.isLoading$$.next(true);
+    return this.courseService.filterCourses(filterInfo).pipe(
+      tap({
+        next: () => {
+          this.isLoading$$.next(false);
+        },
+        error: err => {
+          console.error('Failed to get courses', err);
+          this.isLoading$$.next(false);
+        },
+      })
+    );
   }
 
   getAllAuthors() {
-    // Add your code here
+    this.isLoading$$.next(true);
+    this.courseService.getAllAuthors().subscribe({
+      next: (response: AllAuthorsResponse) => {
+        this.authors$$.next(response.result);
+        this.isLoading$$.next(false);
+      },
+      error: err => {
+        console.error('Failed to get courses', err);
+        this.isLoading$$.next(false);
+      },
+    });
   }
 
-  createAuthor(name: string) {
-    // Add your code here
+  createAuthor(name: Omit<Author, 'id'>) {
+    this.isLoading$$.next(true);
+    this.courseService.createAuthor(name).subscribe({
+      next: (response: CreateAuthorResponse) => {
+        this.authors$$.next([...this.authors$$.value, response.result]);
```

Comment:
Like with courses, authors should be updated as per API response.

Reply:
author is create in a separate page then course so the above mentioned refinement ( see above )  will solve it . When we go back to courses authors and courses will be refreshed.